src/cucumber-runner.ts

```diff
@@ -67,6 +71,48 @@ function buildArgs(runCfg: CucumberRunnerConfig, cucumberBin: string) {
   return procArgs;
 }
 
+/**
+ * Normalizes a Cucumber-js format string.
+ *
+ * For structured inputs (`key:value` or `"key:value"`), returns a string in the
+ * form `"key":"value"`. If the value starts with `file://`, it is treated as an
+ * absolute path and no asset directory is prepended. Otherwise, the asset
+ * directory is prepended to relative paths.
+ *
+ * For simple inputs (e.g., `usage`), the input is returned unchanged.
+ *
+ * @param {string} format - The input format string. Examples include:
+ *                          - `"key:value"`
+ *                          - `"key":"value"`
+ *                          - `key:value`
+ *                          - `usage`
+ * @param {string} assetDir - The directory to prepend to the value for relative paths.
+ * @returns {string} The normalized format string.
+ *
+ * Example:
+ * - Input: `"html":"formatter/report.html"`, `"/project/assets"`
+ *   Output: `"html":"/project/assets/formatter/report.html"`
+ * - Input: `"html":"file://formatter/report.html"`, `"/project/assets"`
```

Comment:
The file url is for the first part, not the second. e.g.

`"file://path/to/formatter/implementation":"/output/file"`

From the docs (edited for clarity):

For each format you specify, you have to provide one or two values. The first (required) is to identify the formatter. It can take a few forms...An absolute path to a local **formatter implementation** in the form of a file:// URL

Reply:
Oh I got what you mean 🤔

Reply:
I think in that case, it's _only_ a path without the key:value structure 🤔 (based on what I'm reading [here](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md#formatters) as well as the [deprecation example](https://github.com/cucumber/cucumber-js/blob/main/docs/deprecations.md#ambiguous-colons-in-formats))

Reply:
Technically speaking, I think it makes sense to have `"file://formatter_implementation":"output_file"`. These two parts are handled separately.

My only concern here is using `"file://formatter_implementation"` as the formatter may potentially cause issue for users. Users must understand the runner's folder structure and specify the exact path, which can be prone to errors, especially with framework upgrades that might break the absolute path.

Reply:
Let me write a test implementation and see if it works 🔍

Reply:
Yeah. Following config works. 
```
--format '"file:///Users/tianfeng/code/saucectl-playwright-example/examples/cucumber/my-formatter.js":"my-formatter.log"'
```
I just printed a message in my-formatter.log.